src/CustomElements/observe.js

```diff
@@ -251,7 +251,11 @@ function takeRecords(node) {
   while (node.parentNode) {
     node = node.parentNode;
   }
-  var observer = node.__observer;
+
+  // this needs to be on head or it will leak in IE
+  // IE does not like it when you have non-standard attributes on root dom's, so put
```

Comment:
My findings are attributes that are not defined in the w3 spec.  I found an example that will cause a leak  as part of the UrlResolver in polymer where a 'a' element is being appended to the root document, this breaks, but if you set the baseURI that does not cause a memory leak.  I think it is because baseURI is defined in the w3 spec while __url is not (https://github.com/Polymer/polymer/pull/3964/files)

a root dom, is any root document that has been created with document.implementation.createHTMLDocument

this is why we need the node.head && node.head.__observer, because placing the observer directly on the node (this is where the code is poorly named because node is always a root shadow document at this point in the code)

we do have test coverage of this, the tests will break if you remove the createElement('head') && createElement('body) that was added Element.js